zerver/lib/create_user.py

```diff
@@ -111,11 +103,25 @@ def create_user_profile(
         tutorial_status=tutorial_status,
         enter_sends=enter_sends,
         onboarding_steps=orjson.dumps([]).decode(),
-        default_language=default_language,
-        twenty_four_hour_time=default_twenty_four_hour_time,
         delivery_email=email,
         **extra_kwargs,
     )
+
+    if realm.string_id != settings.SYSTEM_BOT_REALM:
+        realm_user_default = RealmUserDefault.objects.get(realm=realm)
+
+        display_settings = [
+            setting for setting in UserProfile.property_types.keys() if setting != "timezone"
+        ]
+
+        for settings_name in display_settings:
+            settings_value = getattr(realm_user_default, settings_name)
+            setattr(user_profile, settings_name, settings_value)
+
+        for settings_name in UserProfile.notification_setting_types.keys():
+            settings_value = getattr(realm_user_default, settings_name)
+            setattr(user_profile, settings_name, settings_value)
```

Comment:
We should perhaps add a block comment here making clear how this related to `copy_user_profile`.  I believe the logic is correct -- we want `copy_user_profile` to take precedence over this new feature.  However, it might make sense to refactor this to call `copy_default_settings(default_settings_source: UserBaseSettings)` with either a `UserProfile` or a `RealmUserDefault` function from the same place, which would make it obvious how this works from the code structure.